src/tbm/GiveawayMinter.scilla

```diff
@@ -147,23 +150,46 @@ procedure IsWithinMintLimit(quantity: Uint32)
   end
 end
 
+procedure IsReservedToken(token_id: Uint256)
```

Comment:
is this check necessary? imo use a `distributed_tokens: Uint256`  will do, increment at `DistributeToken` rather then expecting `token_id`. and check for `distributed_tokens` < `reserved_supply` instead